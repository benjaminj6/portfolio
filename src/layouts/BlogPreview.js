// @flow
import React from 'react'

/* eslint-disable */
type Post = {
  title: string,
  preview: string
}
/* eslint-enable */

const PostPreview = ({post}: {post: Post}) => (
  <li>
    <h3>{post.title}</h3>
    <p>
      {post.preview.length > 140
        ? `${post.preview.slice(0, 139)}...`
        : post.preview}
    </p>
  </li>
)

type BlogPreviewProps = {
  headline?: string,
  posts: Array<Post>,
  numPosts?: number
}

const BlogPreview = (props: BlogPreviewProps) => (
  <section>
    <h2>{props.headline || 'Recent posts'}</h2>
    <ul>
      {props.posts
        .slice(0, props.numPosts || 5)
        .map(post => (
          <PostPreview key={`post-${post.title.toLowerCase()}`} post={post} />
        ))}
    </ul>
  </section>
)

export default BlogPreview
