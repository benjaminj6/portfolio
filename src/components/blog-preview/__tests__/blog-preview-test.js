/* eslint-env jest */
import { h } from 'preact'
import { shallow } from 'preact-render-spy'

import BlogPreview from '../'

const posts = [
  {
    publishDate: '2017-10-07',
    title: 'Test Post',
    teaser: 'This is a test post',
    url: 'http://test.com'
  },
  {
    publishDate: '2017-10-09',
    title: 'Test Post #2',
    teaser: 'This is another test post',
    url: 'http://test-2.com'
  }
]

test('blog preview snap', () => {
  const tree = shallow(<BlogPreview posts={posts} />)
  expect(tree).toMatchSnapshot()
})
