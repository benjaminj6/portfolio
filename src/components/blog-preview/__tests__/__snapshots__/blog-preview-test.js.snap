// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`blog preview snap 1`] = `
preact-render-spy (1 nodes)
-------
<div class="jsx-2341525854 jsx-2341525854 BlogPreview">
  <ul class="jsx-2341525854 jsx-2341525854">
    <li class="jsx-2341525854 jsx-2341525854">
      <Card>
        <p class="jsx-2341525854 jsx-2341525854 published">2017-10-07</p>
        <h1 class="jsx-2341525854 jsx-2341525854">
          <A
            href="http://test.com"
            target="_blank"
          >
            Test Post
          </A>
        </h1>
        <p class="jsx-2341525854 jsx-2341525854 teaser">This is a test post...</p>
      </Card>
    </li>
    <li class="jsx-2341525854 jsx-2341525854">
      <Card>
        <p class="jsx-2341525854 jsx-2341525854 published">2017-10-09</p>
        <h1 class="jsx-2341525854 jsx-2341525854">
          <A
            href="http://test-2.com"
            target="_blank"
          >
            Test Post #2
          </A>
        </h1>
        <p class="jsx-2341525854 jsx-2341525854 teaser">This is another test post...</p>
      </Card>
    </li>
  </ul>
  <JSXStyle
    styleId="2341525854"
    css=".jsx-2341525854{--published-color:var(--gray-normal);}li.jsx-2341525854+li.jsx-2341525854{margin-top:1rem;}.published.jsx-2341525854{margin-bottom:1em;color:var(--published-color);font-size:0.75rem;text-align:right;}.teaser.jsx-2341525854{margin:1rem 0 0;}"
  >
  </JSXStyle>
</div>

`;
